src/ida/ida.c

```diff
@@ -3968,69 +3987,31 @@ static int IDARootfind(IDAMem IDA_mem)
  * =================================================================
  */
 
-/*
- * IDAProcessError is a high level error handling function.
- * - If ida_mem==NULL it prints the error message to stderr.
- * - Otherwise, it sets up and calls the error handling function
- *   pointed to by ida_ehfun.
- */
-
-void IDAProcessError(IDAMem IDA_mem, int error_code, const char* module,
-                     const char* fname, const char* msgfmt, ...)
+void IDAProcessError(IDAMem IDA_mem, int error_code, int line, const char* func,
+                     const char* file, const char* msgfmt, ...)
 {
-  va_list ap;
-  char msg[256];
-
   /* Initialize the argument pointer variable
      (msgfmt is the last required argument to IDAProcessError) */
-
+  va_list ap;
   va_start(ap, msgfmt);
 
   /* Compose the message */
-
-  vsprintf(msg, msgfmt, ap);
+  size_t msglen = vsnprintf(NULL, 0, msgfmt, ap) + 1;
+  char* msg     = (char*)malloc(msglen);
+  vsnprintf(msg, msglen, msgfmt, ap);
 
   if (IDA_mem == NULL)
-  { /* We write to stderr */
-#ifndef NO_FPRINTF_OUTPUT
-    fprintf(stderr, "\n[%s ERROR]  %s\n  ", module, fname);
-    fprintf(stderr, "%s\n\n", msg);
-#endif
+  {
+    SUNGlobalFallbackErrHandler(line, func, file, msgfmt, error_code);
   }
   else
-  { /* We can call ehfun */
-    IDA_mem->ida_ehfun(error_code, module, fname, msg, IDA_mem->ida_eh_data);
+  {
+    SUNHandleErrWithMsg(line, func, file, msg, error_code, IDA_mem->ida_sunctx);
   }
 
   /* Finalize argument processing */
   va_end(ap);
-
-  return;
-}
-
-/* IDAErrHandler is the default error handling function.
-   It sends the error message to the stream pointed to by ida_errfp */
-
-void IDAErrHandler(int error_code, const char* module, const char* function,
-                   char* msg, void* data)
-{
-  IDAMem IDA_mem;
-  char err_type[10];
-
-  /* data points to IDA_mem here */
-
-  IDA_mem = (IDAMem)data;
-
-  if (error_code == IDA_WARNING) { sprintf(err_type, "WARNING"); }
```

Comment:
There are couple uses of `IDAProcessError` where the message is intended to be a warning rather than an error.